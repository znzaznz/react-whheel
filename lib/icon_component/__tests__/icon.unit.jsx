import * as renderer from 'react-test-renderer'
import React from 'react'
import Icon from "../icon";
import {mount} from "enzyme";

describe('icon', () => {
    it('Icon 是个svg', () => {
        const json = renderer.create(<Icon name={'weibo'}/>).toJSON()
        expect(json).toMatchSnapshot()
    })
    it('onClick', function () {
        const fn = jest.fn()
        const component = mount(<Icon name="wechat" onClick={fn}/>)
        component.find('svg').simulate('click')
        expect(fn).toBeCalled()
    });
})