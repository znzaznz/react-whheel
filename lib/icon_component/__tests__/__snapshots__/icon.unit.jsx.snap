// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon Icon 是个svg 1`] = `
<svg
  className="react_wheels_icon "
>
  <use
    xlinkHref="#weibo"
  />
</svg>
`;
